SAMPLES = ["A", "B"]

base = "/ldata/1721.account/snakemake_data"


rule all:
    input:
        f"{base}/plots/quals.svg",


rule bwa_map:
    input:
        f"{base}/data/samples/{{sample}}.fastq",
        f"{base}/genome/genome.fa",
        f"{base}/genome/genome.fa.amb",
        f"{base}/genome/genome.fa.ann",
        f"{base}/genome/genome.fa.bwt",
        f"{base}/genome/genome.fa.fai",
        f"{base}/genome/genome.fa.pac",
        f"{base}/genome/genome.fa.sa",
    output:
        "mapped_reads/{sample}.bam",
    shell:
        "mkdir -p mapped_reads && touch mapped_reads/{wildcards.sample}.bam"


rule samtools_sort:
    input:
        "mapped_reads/{sample}.bam",
    output:
        "sorted_reads/{sample}.bam",
    shell:
        "mkdir -p sorted_reads && touch sorted_reads/{wildcards.sample}.bam"


rule samtools_index:
    input:
        "sorted_reads/{sample}.bam",
    output:
        "sorted_reads/{sample}.bam.bai",
    shell:
        "touch sorted_reads/{wildcards.sample}.bam.bai"


rule bcftools_call:
    input:
        fa=f"{base}/genome/genome.fa",
        bam=expand("sorted_reads/{sample}.bam", sample=SAMPLES),
        bai=expand("sorted_reads/{sample}.bam.bai", sample=SAMPLES),
    output:
        "calls/all.vcf",
    shell:
        "mkdir -p calls && touch calls/all.vcf"


rule plot_quals:
    input:
        "calls/all.vcf",
    output:
        f"{base}/plots/quals.svg",
    shell:
        "mkdir -p plots && echo 'abcdefg' > plots/quals.svg && cp plots/quals.svg {output}"
